stdlib/public/core/Duration.swift

```diff
@@ -0,0 +1,228 @@
+    } else {
+      return Int64(bitPattern: bits.low & Bits.nanosecondMask)
+    }
+  }
+  
+  @inlinable
+  public var _sign: (Bool, Bool) {
+    return ((bits.high & Bits.secondsSignFlags) != 0, (bits.low & Bits.nanosecondsSignFlags) != 0)
+  }
+  
+  @inlinable
+  public init(_seconds seconds: Int64, nanoseconds: Int64) {
+    
+    let additionalSeconds = nanoseconds / 1000000000
+    let s = seconds + Int64(additionalSeconds)
+    let ns = nanoseconds - (additionalSeconds * 1000000000)
+    
+    bits = Bits(high: s.magnitude, low: ns.magnitude & Bits.nanosecondMask)
+    if seconds < 0  {
+      bits.low |= Bits.secondsSignFlags
+    }
+    if nanoseconds < 0 {
+      bits.low |= Bits.nanosecondsSignFlags
+    }
+  }
+}
+
+@available(macOS 9999, iOS 9999, tvOS 9999, watchOS 9999, macCatalyst 9999, *)
+extension Duration {
+  @inlinable
+  public static func seconds<T: BinaryInteger>(_ seconds: T) -> Duration {
+    return Duration(_seconds: Int64(seconds), nanoseconds: 0)
+  }
+  
+  @inlinable
+  public static func seconds(_ seconds: Double) -> Duration {
+    let s = Int64(seconds)
+    let ns = Int64((seconds - Double(s)) * 1_000_000_000)
+    return Duration(_seconds: s, nanoseconds: ns)
+  }
+  
+  @inlinable
+  public static func milliseconds<T: BinaryInteger>(_ milliseconds: T) -> Duration {
+    let s = milliseconds / 1000
+    let ns = (milliseconds - s * 1000) * 1000000
+    return Duration(_seconds: Int64(s), nanoseconds: Int64(ns))
+  }
+  
+  @inlinable
+  public static func milliseconds(_ milliseconds: Double) -> Duration {
+    let s = Int64(milliseconds / 1000.0)
+    let ns = Int64((milliseconds - Double(s) * 1000) * 1000000)
+    return Duration(_seconds: s, nanoseconds: ns)
+  }
+  
+  @inlinable
+  public static func microseconds<T: BinaryInteger>(_ microseconds: T) -> Duration {
+    let s = microseconds / 1000000
+    let ns = (microseconds - s * 1000000) * 1000
+    return Duration(_seconds: Int64(s), nanoseconds: Int64(ns))
+  }
+  
+  @inlinable
+  public static func microseconds(_ microseconds: Double) -> Duration {
+    let s = Int64(microseconds / 1000000.0)
+    let ns = Int64((microseconds - Double(s) * 1000000) * 1000)
+    return Duration(_seconds: s, nanoseconds: ns)
+  }
+  
+  @inlinable
+  public static func nanoseconds<T: BinaryInteger>(_ nanoseconds: T) -> Duration {
+    let s = nanoseconds / 1000000000
+    let ns = nanoseconds - s * 1000000000
+    return Duration(_seconds: Int64(s), nanoseconds: Int64(ns))
+  }
+}
+
+@available(macOS 9999, iOS 9999, tvOS 9999, watchOS 9999, macCatalyst 9999, *)
+extension Duration: Codable { }
+
+@available(macOS 9999, iOS 9999, tvOS 9999, watchOS 9999, macCatalyst 9999, *)
+extension Duration: Hashable { }
+
+@available(macOS 9999, iOS 9999, tvOS 9999, watchOS 9999, macCatalyst 9999, *)
+extension Duration: Equatable {
+  @inlinable
+  public static func == (_ lhs: Duration, _ rhs: Duration) -> Bool {
```

Comment:
If there is good reason for `==` to be inlinable, then by symmetry `hash(into:)` should be inlinable too.